### ISSUE TITLE
Registering Interactive Transition for RedirectionCoordinator

### ISSUE THREAD

**Issue by adamsmaka** [quickbirdstudios/XCoordinator#80]
Hey. In my case I have NavigationCoordinator on which I push RedirectionCoordinator.

From this place I would like to apply a gesture driven interactive transition in order to pop back to main super coordinator.

Unfortunately registerInteractiveTransition method from Example project is just an extension of BaseCoordinator and I got an error:

> Use of unresolved identifier 'registerInteractiveTransition'

So how can I achieve custom gesture driven pop interaction transition from RedirectionCoordinator? 

Also I would like to know how to enable back/left swipe gesture for elements of this Navigation Controller stack. Is it possible?

**Comment by pauljohanneskraft** [comment]
When you are pushing the viewController you want a pop-animation on, add an Animation-object including a dismissal-TransitionAnimation. This TransitionAnimation will be executed with the interactivePopGestureRecognizer. You do not need the registerInteractiveTransition-method for this. Thanks for mentioning however, this may be added in a later version.

**Comment by pauljohanneskraft** [comment]
Maybe you're using an older version of XCoordinator? With the most recent version, UINavigationController-animations should be working as expected. Please keep in mind, though, that your rootViewController.delegate should be set by us for this to work, so if you are overriding `generateRootViewController`, please make sure to either set the delegate to the one from the super-call, as in:

```
let rootViewController = MyNavigationController()
rootViewController.delegate = super.generateRootViewController().delegate
return rootViewController
```

Alternatively, you can also create your own `NavigationAnimationDelegate` and set it as its delegate, but keep in mind to hold a strong reference to it.

**Comment by adamsmaka** [comment]
> Maybe you're using an older version of XCoordinator? With the most recent version, UINavigationController-animations should be working as expected.
Yeah you were right. I updated the pod and swipe back works as expected. :) 
Thanks for quick and detailed answer.

Also I would like to allow users swipe from top to bot and allow them to pop with interactive transition. Could you help me how to achieve also that?

Previously I used registerInteractiveTransition like this

`override func presented(from presentable: Presentable?) {
        super.presented(from: presentable)
        addDismissGestureRecognizer(to: rootViewController)
    }
    
    private func addDismissGestureRecognizer(to container: Container) {
        guard let nc = container as? UINavigationController else { return }
        guard let vc = nc.viewControllers.first as? ScrollableViewController else { return }

        let gestureRecognizer = vc.scrollView.panGestureRecognizer
        
        registerInteractiveTransition(
            for: .close,
            triggeredBy: gestureRecognizer,
            progress: { [weak vc] recognizer in
                guard let scrollView = vc?.scrollView else { return 0 }
                let progress = scrollView.contentOffset.y / -200
                return progress
            },
            shouldFinish: { [weak vc] recognizer in
                guard let scrollView = vc?.scrollView else { return false }
                let progress = scrollView.contentOffset.y / -200
                return progress > 0.5
            },
            completion: nil
        )
    }`

But now when I cannot use this method on RedirectionCoordinator there is any other way to enable it?

**Comment by pauljohanneskraft** [comment]
It is unfortunately not available for RedirectionCoordinators, it will be added in a future version! Feel free to add it yourself for now (for example by creating a common superclass for your project). Also feel free to open a Merge Request with your changes :)

**Comment by adamsmaka** [comment]
> It is unfortunately not available for RedirectionCoordinators

That's sad :(

> Feel free to add it yourself for now.

I tried to make an extension for RedirectionCoordinator with registering functions from BaseCoordinator in my project by actually I get a lot of weird error from Xcode because of that. I'm not so pro yet. I don't even know what Merge Request is so I must wait for now.

Thanks anyway!

**Comment by pauljohanneskraft** [comment]
Merge Request is just a different word for Pull Request. I can't work on it right now, so please be a little patient on this. Maybe take a look at how it is implemented in the BaseCoordinator and hopefully you will find a solution for the current situation. If you manage to include this functionality into the RedirectionCoordinator, we'd be grateful for your contribution.

**Comment by pauljohanneskraft** [comment]
We removed RedirectionCoordinator from XCoordinator 2.0 (#122) by making it easier to use multiple coordinators with the same rootViewController, so this should be fixed with the upcoming release. 

We also investigated whether it makes sense to add `registerInteractiveTransition` to RedirectionRouter, which led to the conclusion that this is not easily doable since RedirectionRouter only conform to the `Router` protocol and not to the `Coordinator` protocol. 

We therefore propose that, in the case of a RedirectionRouter, a `ParentRoute` enum case can be used to register the interactive transition with the restriction that the interactive transition can only trigger routes of this parent router (which then might trigger child routes, if needed 😉). This is a rather complicated solution, however we assume that the demand of this feature is rather limited in comparison to the work needed to implement this feature nicely.
